actors/builtin/payment_channel/payment_channel_actor.go

```diff
@@ -16,33 +17,34 @@ import (
 
 type PaymentChannelActor struct{}
 
-/////////////////
-// Constructor //
-/////////////////
-
-type PCAConstructorParams struct {
+type ConstructorParams struct {
 	To addr.Address
 }
 
-func (pca *PaymentChannelActor) Constructor(rt vmr.Runtime, params *PCAConstructorParams) *adt.EmptyValue {
-
-	// TODO anorth: ensure parties are valid
-	// rt.ValidateImmediateCallerType(builtin.CallerTypesSignable...)
-	// toID, err := addr.IDFromAddress(params.To)
-	// if err != nil {
-	// 	rt.Abort(exitcode.ErrIllegalArgument, "receiver must be ID address")
-	// }
+func (pca *PaymentChannelActor) Constructor(rt vmr.Runtime, params *ConstructorParams) *adt.EmptyValue {
+	rt.ValidateImmediateCallerType(builtin.AccountActorCodeID)
+
+	// Check that the channel creator is capable of signing vouchers.
+	creator := rt.ImmediateCaller()
+	ret, code := rt.Send(creator, builtin.MethodsAccount.PubkeyAddress, &adt.EmptyValue{}, big.Zero())
+	builtin.RequireSuccess(rt, code, "failed to request pubkey address for %v", creator)
+	var signingAddress addr.Address
+	err := ret.Into(&signingAddress)
+	if err != nil {
+		rt.Abort(exitcode.ErrSerialization, "failed to deserialize address %v", ret)
+	}
+	if signingAddress.Protocol() != addr.SECP256K1 && signingAddress.Protocol() != addr.BLS {
+		rt.Abort(exitcode.ErrIllegalArgument, "creator's signing address must use SECP or BLS protocol, was %v", signingAddress.Protocol())
+	}
 
-	// if !toID.PubKeyAddress.Protocol() == addr.SECP256K1 && !toID.PubKeyAddress.Protocol() == addr.BLS {
-	// 	rt.Abort(exitcode.ErrIllegalArgument, "address must use BLS or SECP protocol, got %v", toID.PubKeyAddress.Protocol())
-	// }
+	// Check that target is a canonical ID address.
+	// This is required for consistent caller validation.
```

Comment:
must also check for from then since either party can call methods on this (this may implictly be done above already)

Reply:
Yes, implicit, `rt.ImmediateCaller()` is always an ID address.